observing/schedule.py

```diff
@@ -60,6 +61,32 @@ def get_sched():
     return dd, dd2
 
 
+def check_sched(sched):
+    """ Check if sched is requesting a beam that is already scheduled or submitted
+    """
+
+    dd, dd2 = get_sched()
+    sched_dict = create_dict(sched)
+
+    ok = True
+    for kk,vv in sched_dict.items():
+        if kk in dd.keys():
+            for tr in dd[kk].values():
+                _, (t0, t1) = vv.popitem()
```

Comment:
In general, don't modify when you're in a `for` loop.

Reply:
I don't think it is, but it I agree it is messy. Working on that.

Reply:
`popitem` does remove.

This is a good example of the serialize/deserialize madness that can happen. You know the data structure, but then when you go back and forth it turns into a mess.

If you can specify an `object_hook` in `json.dumps` that turns this json into a legible data structure, that'd be better.

See if https://docs.python.org/3/library/types.html#types.SimpleNamespace works but I don't know if it does nested namespace well.

Otherwise, you may have to implement your own serializable class (I actually don't know what a good python way to do this is but), so like

```
class schedule:
  def to_json(self):
       # try using self.__dict__ here
  
  def from_json(cls):
      # idk

def demo():
    json.loads(some_json_string, object_hook=schedule.from_json)
    json.dumps(some_schedule_object.to_json()))
```